lib/callback_model.js

```diff
@@ -24,6 +24,10 @@ class CallbackModel extends EventEmitter {
     this.connection.close(cb);
   }
 
+  updateSecret(newSecret, reason, cb) {
+    this.connection._updateSecret(newSecret, reason, cb);
+  }
+
```

Comment:
I was in two minds whether to default reason, but decided against it as I wasn't sure what a sensible default value would be. Happy to reconsider.